src/input/commodity/levy.rs

```diff
@@ -114,15 +115,46 @@ where
         }
     }
 
-    // Validate map
+    // Validate map and complete with missing regions/years/time slices
     for (commodity_id, regions) in commodity_regions.iter() {
-        let map = map.get(commodity_id).unwrap();
+        let map = map.get_mut(commodity_id).unwrap();
         validate_commodity_levy_map(map, regions, milestone_years, time_slice_info)
             .with_context(|| format!("Missing costs for commodity {commodity_id}"))?;
+
+        let difference: IndexSet<_> = region_ids.difference(regions).cloned().collect();
+        for region_id in difference.iter() {
```

Comment:
You can just iterate over this directly so you don't have to store anything:

```suggestion
        for region_id in region_ids.difference(regions) {
```